cardano-db/src/Cardano/Db/Migration/Extra/CosnumedTxOut/Queries.hs

```diff
@@ -202,40 +222,114 @@ countConsumed = do
     pure countRows
   pure $ maybe 0 unValue (listToMaybe res)
 
-deleteConsumedTxOut :: forall m. MonadIO m => Trace IO Text -> Word64 -> ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut ::
+  forall m.
+  (MonadIO m, MonadBaseControl IO m) =>
+  Trace IO Text ->
+  Word64 ->
+  ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut trce blockNoDiff = do
+  maxTxInId <- findMaxTxInId blockNoDiff
+  case maxTxInId of
+    Left _ -> do
+      createConsumedTxOut
+      liftIO $ logInfo trce "Created ConsumedTxOut as there were no blocks present"
+    Right mxtxid -> do
+      migrateNextPage mxtxid False 0
+  where
+    migrateNextPage :: TxInId -> Bool -> Word64 -> ReaderT SqlBackend m ()
+    migrateNextPage mxTxInId ranCreateConsumedTxOut offst = do
+      pageEntries <- getInputPage offst pageSize
+      resPageEntries <- splitAndProcessPageEntries trce ranCreateConsumedTxOut mxTxInId pageEntries
+      when (fromIntegral (length pageEntries) == pageSize) $
+        migrateNextPage mxTxInId resPageEntries $!
+          offst
+            + pageSize
+
+--
+splitAndProcessPageEntries ::
+  forall m.
+  (MonadIO m, MonadBaseControl IO m) =>
+  Trace IO Text ->
+  Bool ->
+  TxInId ->
+  [(TxInId, TxId, Word64)] ->
+  ReaderT SqlBackend m Bool
+splitAndProcessPageEntries trce ranCreateConsumedTxOut maxTxInId entries = do
+  let entriesSplit = span (\(txInId, _, _) -> txInId <= maxTxInId) entries
+  case entriesSplit of
+    -- empty lists just return
+    ([], []) -> pure True
+    -- the whole list is less that maxTxInId
+    (xs, []) -> do
+      deleteEntries xs
+      pure False
+    -- the whole list is greater that maxTxInId
+    ([], ys) -> do
+      shouldCreateConsumedTxOut ranCreateConsumedTxOut
+      updateEntries ys
+      pure True
+    -- the list has both bellow and above maxTxInId
+    (xs, ys) -> do
+      deleteEntries xs
+      shouldCreateConsumedTxOut ranCreateConsumedTxOut
+      updateEntries ys
+      pure True
+  where
+    -- the delete + update can happen as one query
+    deleteEntries = mapM_ (\(_, txId, index) -> deleteTxOutConsumed txId index)
+    updateEntries = mapM_ (\(txInId, txId, index) -> updateTxOutConsumedByTxInIdUnique txId index txInId)
```

Comment:
@kderme I've isolated the place we could bulk delete/update but I'm not 100% how to do that in persistent when it comes to `deleteWhere`

Reply:
I guess it could work like 
```
deleteWhere [FilterOr <$> mkFilter xs]
```
or by using `IN` clause.
But that's great already, it makes it easy to try different queries during benchmarking.